src/Compilers/CSharp/Portable/Symbols/PublicModel/TypeSymbol.cs

```diff
@@ -168,5 +169,14 @@ ImmutableArray<SymbolDisplayPart> ITypeSymbol.ToMinimalDisplayParts(SemanticMode
         bool ITypeSymbol.IsRefLikeType => UnderlyingTypeSymbol.IsRefLikeType;
 
         bool ITypeSymbol.IsReadOnly => UnderlyingTypeSymbol.IsReadOnly;
+
+        bool ITypeSymbol.IsRecord
+        {
+            get
+            {
+                HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+                return SynthesizedRecordClone.FindValidCloneMethod(UnderlyingTypeSymbol, ref useSiteDiagnostics) != null;
```

Comment:
This is likely not going to be resilient to future `with` expression changes. I do not feel comfortable basing the public API on this implementation detail